business/definitionService.js

```diff
@@ -176,22 +179,76 @@ class DefinitionService {
     )
   }
 
-  /**
-   * Compute facets related to an entity and a curation of that entity. This is essentially a light
-   * weight pre-compute so we know what groupings to use in the real summarization.
-   *
-   * @param {EntitySpec} coordinates - The entity for which we are looking for a curation
-   * @param {*} raw - the set of raw tool ouptuts related to the idenified entity
-   * @param {Definition)} [curation] - an actual curation object to apply.
-   * @returns {Facets} The `facets` portion of the component's described definition
-   */
-  async _computeFacets(coordinates, raw, curation) {
-    // TODO we might be able to a more effecient computation here since all we need are the facets
-    // for example, some tool outputs just will never have facets to there is no point in summarizing them.
-    const summarized = await this.summaryService.summarizeFacets(coordinates, raw)
-    const aggregated = await this.aggregationService.process(coordinates, summarized)
-    const definition = await this.curationService.apply(coordinates, curation, aggregated)
-    return get(definition, 'described.facets')
+  // ensure all the right facet information has been computed and added to the given definition
+  _ensureFacets(definition) {
+    if (!definition.files) return
+    const facetFiles = this._computeFacetFiles([...definition.files], definition.described.facets)
+    for (const facet in facetFiles)
+      setIfValue(definition, `licensed.facets.${facet}`, this._summarizeFacetInfo(facet, facetFiles[facet]))
```

Comment:
I wonder if it would simplify to call `_summarizeFacetInfo` within the loop in `_computeFacetFiles` since it modifies the facets in place

so it would look something like

```
_computeFacetFiles(files, facets = {}) {
    const facetList = Object.getOwnPropertyNames(facets)
    remove(facetList, 'core')
    if (facetList.length === 0) return { core: files }
    const result = { core: [...files] }
    for (const facet in facetList) {
      const facetKey = facetList[facet]
      const filters = facets[facetKey]
      if (!filters || filters.length === 0) break
      const facetFile = files.filter(file => filters.some(filter => minimatch(file.path, filter)));
      result[facetKey] = this._summarizeFacetInfo(facetKey, facetFile)
      pullAllWith(result.core, result[facetKey], isEqual)
    }
    return result
  }
```